google/cloud/forseti/scanner/scanners/base_scanner.py

```diff
@@ -102,24 +104,14 @@ def _output_results_to_db(self, violations):
 
         Args:
             violations (list): A list of violations.
-
-        Returns:
-            list: Violations that encountered an error during insert.
         """
         model_description = (
             self.service_config.model_manager.get_description(self.model_name))
         inventory_index_id = (
             model_description.get('source_info').get('inventory_index_id'))
 
-        # TODO: Capture violations errors with the new violation_access.
-        # Add a unit test for the errors.
-        (inserted_row_count, violation_errors) = (0, [])
-
-        violation_access = self.service_config.violation_access(
-            self.service_config.engine)
-        violation_access.create(violations, inventory_index_id)
-        # TODO: figure out what to do with the errors. For now, just log it.
-        LOGGER.debug('Inserted %s rows with %s errors',
-                     inserted_row_count, len(violation_errors))
-
-        return violation_errors
+        violation_access = self.service_config.violation_access
+        scanner_index_id = scanner_dao.get_latest_scanner_id(
+            violation_access.session, inventory_index_id,
+            index_state=IndexState.RUNNING)
```

Comment:
This needs to be more specific here.  Could there be other `RUNNING` states due to previous crashes?  i.e. is `RUNNING` always guaranteed to be 1?

Reply:
There could be other `RUNNING` scanner index instances but we are getting the latest instance here. IOW, we are fine.